server/chatroom.py

```diff
@@ -20,36 +23,188 @@ def get_msg_update(cur, chatroom_id, date_time):
+        )
+        
+        cur.execute("INSERT INTO invite values(%(invite_id)s, %(chatroom_id)s, %(invite_sender_id)s, %(target_user_id)s, %(uni_id)s)",
+                    {"invite_id" : invite_id, "chatroom_id" : chatroom_id, "invite_sender_id" : invite_sender_id, "target_user_id" : target_user_id, "uni_id" : uni_id})
+        
+
         return True
-    except:
-        return False
+
+    except BaseException as e:
+        print(f'Exception: {e}')
+        raise HTTPException(
+            status_code=500,
+            detail="Database Error",
+        )
+
+#Accept an invite and join a (non-course) chatroom
+def accept_invite(cur, invite_id, target_user_id):
+    try:
+        #make sure the target user is verified
+        cur.execute("SELECT student_id FROM student WHERE student_id = %(target_user_id)s",
+                    {"target_user_id" : target_user_id})
+        res = cur.fetchall()[0][0]
+        if (res != target_user_id):
+            raise HTTPException(status_code=500, detail="Student not verified!")
+        
+        #get target user of invite, check to see if it matches target_user_id parameter
+        cur.execute("SELECT target_user_id FROM invite WHERE invite_id=%(invite_id)s", {"invite_id" : invite_id})
+        res = cur.fetchall()[0][0]
+        if (res != target_user_id):
+            raise HTTPException(status_code=500, detail="The invite is not for this user")
+
+        
+        #cur.execute("SELECT id FROM chatroom WHERE invite_id = %(invite_id)s", {"invite_id" : invite_id})
+        cur.execute("SELECT chatroom_id FROM invite WHERE invite_id=%(invite_id)s", {"invite_id" : invite_id})
+        chatroom_id = cur.fetchall()[0][0]
+        
+        #do not allow insertion of user into in_chatroom if uni_id for target user and chatroom do not match
+        cur.execute("SELECT uni_id FROM chatroom WHERE id = %(chatroom_id)s", {"chatroom_id" : chatroom_id})
+        uni_id = cur.fetchall()[0][0]
+        cur.execute("SELECT uni_id FROM student WHERE student_id = %(target_user_id)s", {"target_user_id" : target_user_id})
+        
+        student_uni_id = cur.fetchall()[0][0]
+        
+        if (student_uni_id != uni_id):
+            raise HTTPException(status_code=500, detail="The university of the invitee does not match the university of the chatroom")
+        
+        #insert target user into in_chatroom
+        cur.execute("INSERT INTO in_chatroom(student_id, uni_id, chatroom_id) VALUES(%(target_user_id)s, %(uni_id)s, %(chatroom_id)s)",
+                    {"target_user_id" : target_user_id, "uni_id" : uni_id, "chatroom_id" : chatroom_id})
+        
+        return True
+    
+    except BaseException as e:
+        print(f'Exception: {e}')
+        raise HTTPException(
+            status_code=500,
+            detail="Database Error",
+        )
+
+# Get the list of chatrooms that a student is in (for visibility of chatrooms functionality)
+def get_chatrooms_for_student(cur, student_id):
```

Comment:
get_chatroom is already a function in main. please rebase and make changes in get_chatroom instead